compiler/rustc_lint/src/context.rs

```diff
@@ -952,6 +952,18 @@ pub trait LintContext: Sized {
                     db.span_label(first_reexport_span, format!("the name `{}` in the {} namespace is first re-exported here", name, namespace));
                     db.span_label(duplicate_reexport_span, format!("but the name `{}` in the {} namespace is also re-exported here", name, namespace));
                 }
+                BuiltinLintDiagnostics::AmbiguousUpdatedGlob { label_span, label_msg, note_msg, b1_span, b1_note_msg, b1_help_msgs, b2_span, b2_note_msg, b2_help_msgs, .. } => {
+                    db.span_label(label_span, label_msg);
+                    db.note(note_msg);
+                    db.span_note(b1_span, b1_note_msg);
+                    for help_msg in b1_help_msgs {
+                        db.help(help_msg);
+                    }
+                    db.span_note(b2_span, b2_note_msg);
+                    for help_msg in b2_help_msgs {
+                        db.help(help_msg);
+                    }
```

Comment:
All this logic can be shared between `fn report_ambiguity_error` and this place.
(It can be put into a method on `AmbiguityErrorDiag`, for example.)

Reply:
This comment wasn't addressed.

Reply:
> shared between fn report_ambiguity_error and this place.

This poses a small issue because `﻿AmbiguityErrorDiag` is defined in ﻿rustc_lint_defs and cannot directly use ﻿`DiagnosticBuilder`. To resolve this, we would need to introduce additional dependencies for ﻿`rustc_lint_defs`. Do you believe it is worthwhile to do so?